chore(logs): update SDK onboarding pages

JoshuaMoelans opened the pull request:
Continuing from https://github.com/getsentry/sentry/pull/102398, to be merged after that one was merged. 

We add the `logs` onboarding for each of the platforms that now support logs (unity, unreal, dotnet, native, php-symfony)

ToDo
- [x] add Unity onboarding
- [x] double-check that all relevant onboarding docs exist & are correct (I will ping the SDK maintainers)
- [x] maybe add `logs` checkbox to project onboarding page for each of these SDKs (see [original comment](https://github.com/getsentry/sentry/pull/102398#discussion_r2517834592))

JoshuaMoelans committed:
add to withLoggingOnboarding

JoshuaMoelans committed:
add native logs onboarding

JoshuaMoelans committed:
Add logs onboarding for dotnet, symfony and unreal

JoshuaMoelans committed:
add tracing to native/unity/unreal

JoshuaMoelans committed:
fixup native onboarding
- fix link
- add minidump storage setting

JoshuaMoelans committed:
add mock for native onboarding

JoshuaMoelans committed:
use DocsParams directly

JoshuaMoelans committed:
chore(cleanup): remove tracing-related content to keep original PR logs-only (#105563)

JoshuaMoelans committed:
further performace_monitoring cleanup

JoshuaMoelans committed:
add/fix logs onboarding

priscilawebdev commented on static/app/gettingStartedDocs/unreal/logs.tsx:
```diff
@@ -0,0 +1,100 @@
+          ),
+        },
+      ],
+    },
```
we currently repeat this exact text in `onboarding.tsx`. I’m wondering if it would be better to move it to `unreal/utils.tsx` (needs to be created) and reuse it. That would make maintenance easier.

priscilawebdev commented on static/app/gettingStartedDocs/unity/onboarding.tsx:
```diff
@@ -57,6 +58,19 @@ export const onboarding: OnboardingConfig = {
+              type: 'text',
+              text: tct(
+                'To enable structured logging, check the [code:Enable Logs] option in the Sentry configuration window.',
+                {code: <code />}
```
do we need this? I notice that the code snippet already gets updated when "logs" is checked

priscilawebdev commented on static/app/gettingStartedDocs/unreal/logs.tsx:
```diff
@@ -0,0 +1,100 @@
+        {
+          type: 'code',
+          language: 'cpp',
+          code: `USentrySubsystem* SentrySubsystem = GEngine->GetEngineSubsystem<USentrySubsystem>();
```
Rather than adding a separate section just for logs, shouldn’t we update the verify code snippet to include the log’s code? javascript-react is doing that for example

priscilawebdev commented on static/app/components/onboarding/productSelection.tsx:
```diff
@@ -92,7 +92,11 @@ export const platformProductAvailability = {
+    ProductSolution.PROFILING,
+    ProductSolution.LOGS,
+  ],
   'dotnet-aspnet': [ProductSolution.PERFORMANCE_MONITORING],
```
are we going to add LOGS here and other platforms afterwards?  Just noticed that it's "new" in our docs https://docs.sentry.io/platforms/dotnet/guides/aspnet/logs/

JoshuaMoelans replied:
added in https://github.com/getsentry/sentry/pull/104234/changes/cc831e2de2c183b43e768964e3e511d72d66ff5d

alexander-alderman-webb committed:
feat(onboarding): Add logs for .NET platforms

alexander-alderman-webb committed:
add comma to gcp config

alexander-alderman-webb committed:
feat(onboarding): Logs page for .NET platforms (#106550)

<!-- Describe your PR here. -->

<!--

  Sentry employees and contractors can delete or ignore the following.

-->

Look, I get it. The entity doing business as "Sentry" was incorporated
in the State of Delaware in 2015 as Functional Software, Inc. and is
gonna need some rights from me in order to utilize my contributions in
this here PR. So here's the deal: I retain all rights, title and
interest in and to my contributions, and by keeping this boilerplate
intact I confirm that Sentry can use, modify, copy, and redistribute my
contributions, under Sentry's choice of terms.

JoshuaMoelans committed:
cleanup unreal gettingStartedDocs

JoshuaMoelans committed:
remove unnecessary link from native logs

JoshuaMoelans committed:
reduce duplication for Unreal logs

JoshuaMoelans committed:
logsVerify for native

JoshuaMoelans committed:
verifyLogs for Unity

JoshuaMoelans committed:
cleanup unreal onboarding

JoshuaMoelans committed:
logsVerify for php-symfony

JoshuaMoelans committed:
fix test

Flash0ver reviewed (approved):
✅ `dotnet` platforms

JoshuaMoelans committed:
small unity fix

JoshuaMoelans committed:
add Godot logs

JoshuaMoelans committed:
fix Unity API

JoshuaMoelans committed:
update godot logs

priscilawebdev commented on static/app/gettingStartedDocs/dotnet/logs.tsx:
```diff
@@ -0,0 +1,105 @@
+  ],
+});
+
+export const dotnetLogs = (): OnboardingConfig => ({
```
what about calling it just logs.? also... does it need to be a function?

JoshuaMoelans committed:
bump dotnet version fallback & simplify fallback version in utils.tsx

Flash0ver commented on static/app/gettingStartedDocs/dotnet/utils.tsx:
```diff
@@ -1,16 +1,26 @@
-)}`;
+  return `
+Install-Package Sentry -Version ${getPackageVersion(params, 'sentry.dotnet', version)}`;
+};
```
After a chat: it takes the latest non-pre-release from `sentry-release-registry`, and uses the passed version as a fallback. As @JoshuaMoelans suggested, yes, let's indeed just use the currently latest version as a fallback ... it has the most features and optimizations.

priscilawebdev commented on static/app/gettingStartedDocs/dotnet/onboarding.tsx:
```diff
@@ -14,14 +15,14 @@ const getInstallProfilingSnippetPackageManager = (params: DocsParams) => `
   params,
   'sentry.dotnet.profiling',
-  '4.3.0'
+  '6.0.0'
```
is this updating version for profile? if yes, then perhaps it would be better to have this update in a separate PR

JoshuaMoelans committed:
cleanup unnecessary godot logs code snippet

JoshuaMoelans committed:
update dotnetLogs to logs

JoshuaMoelans committed:
change logs tests to be visual

JoshuaMoelans committed:
fix godot param

JoshuaMoelans committed:
fix Xamarin conditional init

JoshuaMoelans commented on static/app/gettingStartedDocs/dotnet-xamarin/onboarding.tsx:
```diff
@@ -35,6 +36,8 @@ public class MainActivity : global::Xamarin.Forms.Platform.Android.FormsAppCompa
             // We recommend adjusting this value in production.
             options.TracesSampleRate = 1.0;
+            // Enable logs to be sent to Sentry
+            options.EnableLogs = true;
```
resolved in https://github.com/getsentry/sentry/pull/104234/commits/154445da5509411a0ec811b2a4b178852750de2c

JoshuaMoelans committed:
add gettingStartedDocs passthrough for cocoa-objc/cocoa-swift

priscilawebdev committed:
improve logs tests

priscilawebdev commented on static/app/gettingStartedDocs/dotnet/logs.spec.tsx:
```diff
@@ -0,0 +1,39 @@
+    expect(verifySteps[0].content).toHaveLength(1);
+    expect(verifySteps[0].content[0].type).toBe('conditional');
+    const conditionalContent = verifySteps[0].content[0].content;
+    expect(conditionalContent[1].code).toContain('SentrySdk.Logger.LogInfo');
```
Thanks for updating the tests! I’d suggest placing them next to each location where the Logs instructions have their own file.

I went ahead and pushed a commit that improves the tests - hope that is fine! please let me know if you have any questions:
https://github.com/getsentry/sentry/pull/104234/changes/f5f2f36d3f2f347d636c9258644b4b3a0c9e0f64

priscilawebdev commented:
Just a quick piece of feedback, maybe for a follow-up.

I noticed that .NET platforms don’t include the common Verify step. We currently only show Verify Logs, and if Traces is selected, we don’t provide an example for verifying traces. Other platforms (for example, React) do include this, so I was wondering if it would make sense to consolidate the behavior across platforms.


https://github.com/user-attachments/assets/825392e6-06c8-4bce-b206-2731e52094f3

priscilawebdev commented on static/app/gettingStartedDocs/dotnet-aspnetcore/onboarding.spec.tsx:
```diff
@@ -40,4 +42,14 @@ describe('aspnetcore onboarding docs', () => {
+    });
+
+    expect(
+      await screen.findByText(textWithMarkupMatcher(/o.EnableLogs/))
```
same here

priscilawebdev reviewed (approved):
Approving since it’s been a while and providing logs during onboarding is valuable, though I do think the logs/Verify steps need to be consolidated. Please let's polish this in a follow up 🙏

priscilawebdev committed:
Merge branch 'master' into joshua/chore/logs_onboarding

JoshuaMoelans committed:
add godot to platformProductAvailability
